File: gunhoflash/maximum-subarray.py
from typing import List

class Solution:
  def maxSubArray(self, nums: List[int]) -> int:
    sumFromStart = 0
    minSum = 0
    maxSum = nums[0]

    for num in nums:
      sumFromStart += num
      maxSum = max(maxSum, sumFromStart - minSum)
      minSum = min(minSum, sumFromStart)

    return maxSum

File: gunhoflash/maximum-subarray.ts
function maxSubArray(nums: number[]): number {
  let sumFromStart = 0;
  let minSum = 0;
  let maxSum = nums[0];

  for (const num of nums) {
    sumFromStart += num;
    maxSum = Math.max(maxSum, sumFromStart - minSum);
    minSum = Math.min(minSum, sumFromStart);
  }

  return maxSum;
}
